Mathlib/Analysis/Meromorphic/TrailingCoefficient.lean

```diff
@@ -0,0 +1,245 @@
+
+/-!
+## Behavior under Arithmetic Operations
+-/
+
+/--
+The trailing coefficient of a scalar product is the scalar product of the trailing coefficients.
+-/
+lemma MeromorphicAt.meromorphicTrailingCoeffAt_smul {f₁ : 𝕜 → 𝕜} {f₂ : 𝕜 → E}
+    (hf₁ : MeromorphicAt f₁ x) (hf₂ : MeromorphicAt f₂ x) :
+    meromorphicTrailingCoeffAt (f₁ • f₂) x =
+      (meromorphicTrailingCoeffAt f₁ x) • (meromorphicTrailingCoeffAt f₂ x) := by
+  by_cases h₁f₁ : meromorphicOrderAt f₁ x = ⊤
+  · simp_all [hf₁, hf₁.smul hf₂, meromorphicOrderAt_smul hf₁ hf₂, h₁f₁]
+  by_cases h₁f₂ : meromorphicOrderAt f₂ x = ⊤
+  · simp_all [hf₁, hf₁.smul hf₂, meromorphicOrderAt_smul hf₁ hf₂, h₁f₁]
+  obtain ⟨g₁, h₁g₁, h₂g₁, h₃g₁⟩ := (meromorphicOrderAt_ne_top_iff hf₁).1 h₁f₁
+  obtain ⟨g₂, h₁g₂, h₂g₂, h₃g₂⟩ := (meromorphicOrderAt_ne_top_iff hf₂).1 h₁f₂
+  have : f₁ • f₂ =ᶠ[𝓝[≠] x]
+      fun z ↦ (z - x) ^ (meromorphicOrderAt (f₁ • f₂) x).untop₀ • (g₁ • g₂) z := by
+    filter_upwards [h₃g₁, h₃g₂, self_mem_nhdsWithin] with y h₁y h₂y h₃y
+    simp_all [meromorphicOrderAt_smul hf₁ hf₂]
+    rw [← smul_assoc, ← smul_assoc, smul_eq_mul, smul_eq_mul, zpow_add₀ (sub_ne_zero.2 h₃y)]
+    ring_nf
+  rw [h₁g₁.meromorphicTrailingCoeffAt_of_order_eq_finite₁ h₂g₁ h₃g₁,
+    h₁g₂.meromorphicTrailingCoeffAt_of_order_eq_finite₁ h₂g₂ h₃g₂,
+    (h₁g₁.smul h₁g₂).meromorphicTrailingCoeffAt_of_presentation this]
+  simp
+
+/--
+The trailing coefficient of a product is the product of the trailing coefficients.
+-/
+lemma MeromorphicAt.meromorphicTrailingCoeffAt_mul {f₁ f₂ : 𝕜 → 𝕜} (hf₁ : MeromorphicAt f₁ x)
+    (hf₂ : MeromorphicAt f₂ x) :
+    meromorphicTrailingCoeffAt (f₁ * f₂) x =
+      (meromorphicTrailingCoeffAt f₁ x) * (meromorphicTrailingCoeffAt f₂ x) :=
+  meromorphicTrailingCoeffAt_smul hf₁ hf₂
+
+/--
+The trailing coefficient of the inverse function is the inverse of the trailing coefficient.
+-/
+lemma meromorphicTrailingCoeffAt_inv {f : 𝕜 → 𝕜} :
+    meromorphicTrailingCoeffAt f⁻¹ x = (meromorphicTrailingCoeffAt f x)⁻¹ := by
+  by_cases h₁ : MeromorphicAt f x
+  · by_cases h₂ : meromorphicOrderAt f x = ⊤
+    · simp_all [meromorphicOrderAt_inv (f := f) (x := x)]
+    have : f⁻¹ * f =ᶠ[𝓝[≠] x] 1 := by
+      filter_upwards [(meromorphicOrderAt_ne_top_iff_eventually_ne_zero h₁).1 h₂]
+      simp_all
+    rw [← mul_eq_one_iff_eq_inv₀ (h₁.meromorphicTrailingCoeffAt_ne_zero h₂),
+      ← h₁.inv.meromorphicTrailingCoeffAt_mul h₁, meromorphicTrailingCoeffAt_congr_nhdNE this,
+      analyticAt_const.meromorphicTrailingCoeffAt_of_order_eq_finite₁ (n := 0)]
+    <;> simp_all
+    exact eventuallyEq_nhdsWithin_of_eqOn fun _ ↦ congrFun rfl
+  · simp_all
+
+/--
+The trailing coefficient of the power of a function is the power of the trailing coefficient.
+-/
+lemma MeromorphicAt.meromorphicTrailingCoeffAt_zpow {f : 𝕜 → 𝕜} (h₁ : MeromorphicAt f x) :
+    meromorphicTrailingCoeffAt (f ^ n) x = (meromorphicTrailingCoeffAt f x) ^ n := by
```

Comment:
When one arrives here, it is a little bit hard to guess what `n` is, because you haven't used it a lot. I would remove it from the `variable` line, and include it in the statements that need it (only this one?)